### PR TITLE
Accounts integration

### PR DESCRIPTION
**Pull request by fjl**
With this PR, transactions on the JS REPL will prompt for the passphrase to unlock the sender account
while the transaction is being sent.

Mist integration and support for unlocking accounts for the RPC server will happen in a followup PR.

**Note:** the password prompting seems to be a bit unreliable and does echo the password sometimes.

### CODE REVIEW

File: cmd/mist/gui.go [hunk 1]
```diff
@@ -172,7 +160,11 @@ func (gui *Gui) showWallet(context *qml.Context) (*qml.Window, error) {
 	return gui.win, nil
 }
 
-func (gui *Gui) ImportKey(filePath string) {
+func (gui *Gui) GenerateKey() {
+	_, err := gui.eth.AccountManager().NewAccount("hurr")
```
Comment (Gustav-Simonsson) [inline cmd/mist/gui.go]: maybe use empty string as passphrase here for now?

File: xeth/xeth.go [hunk 2]
```diff
@@ -40,37 +38,62 @@ type Backend interface {
...
+	//
+	// It should unlock the account with the given address and return
+	// true if unlocking succeeded.
+	UnlockAccount(address []byte) bool
+
+	// This is called for all transactions inititated through
+	// Transact. It should prompt the user to confirm the transaction
+	// and return true if the transaction was acknowledged.
+	//
+	// ConfirmTransaction is not used for Call transactions
```
Comment (Gustav-Simonsson) [inline xeth/xeth.go]: Why cannot Call txs change state?

File: xeth/xeth.go [hunk 3]
```diff
@@ -304,25 +327,61 @@ func (self *XEth) Transact(toStr, valueStr, gasStr, gasPriceStr, codeStr string)
...
 	}
-
 	return toHex(tx.Hash()), nil
 }
+
+func (self *XEth) sign(tx *types.Transaction, from []byte, didUnlock bool) error {
+	sig, err := self.accountManager.Sign(accounts.Account{Address: from}, tx.Hash())
+	if err == accounts.ErrLocked {
+		if didUnlock {
+			return fmt.Errorf("sender account still locked after successful unlock")
```
Comment (Gustav-Simonsson) [inline xeth/xeth.go]: Why can this happen? Race? If it's an expected race then the error message should be a bit different, e.g. "account unlocked, please unlock again".

**Comment by Gustav-Simonsson** [comment]
blocktests fail with:
```bash
HEAD is now at 80985f9... cmd/evm, core, ethdb, state, tests/helper: remove ReadConfig calls
00:59:48 ((detached from felix/accounts-integration)) ~/go/src/github.com/ethereum/go-ethereum/cmd/blocktest
> go build && ./blocktest /home/gs/eth/tests/BlockTests/bcValidBlockTest.json 
2015/03/10 00:59:55 [STATE] (+) e4157b34ea9615cfbde6b4fda419828124b70c78
2015/03/10 00:59:55 [STATE] (+) b9c015918bdaba24b4ff057a92a3873d6eb201be
2015/03/10 00:59:55 [STATE] (+) cd2a3d9f938e13cd947ec05abc7fe734df8dd826
2015/03/10 00:59:55 [STATE] (+) e6716f9544a56c530d868e4bfbacb172315bdead
2015/03/10 00:59:55 [STATE] (+) 1a26338f0d905e295fccb71fa9ea849ffa12aaf4
2015/03/10 00:59:55 [STATE] (+) 0000000000000000000000000000000000000003
2015/03/10 00:59:55 [STATE] (+) 0000000000000000000000000000000000000002
2015/03/10 00:59:55 [STATE] (+) 0000000000000000000000000000000000000004
2015/03/10 00:59:55 [STATE] (+) dbdbdb2cbd23b783741e8d7fcf51e459b497e4a6
2015/03/10 00:59:55 [STATE] (+) 6c386a4b26f73c802f34673f7248bb118f97424a
2015/03/10 00:59:55 [STATE] (+) 2ef47100e0787b915105fd5e3f4ff6752079d5cb
2015/03/10 00:59:55 [STATE] (+) 0000000000000000000000000000000000000001
2015/03/10 00:59:55 [CHAIN] Last block (#0) b5d6d8402156c5c1dfadaa4b87c676b5bcadb17ef9bc8e939606daaa0d35f55d TD=0
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x510645]

goroutine 1 [running]:
github.com/ethereum/go-ethereum/core.(*ChainManager).InsertChain(0xc2080e8420, 0xc2080aded0, 0x1, 0x1, 0x0, 0x0)
	/home/gs/go/src/github.com/ethereum/go-ethereum/core/chain_manager.go:410 +0x145
main.main()
	/home/gs/go/src/github.com/ethereum/go-ethereum/cmd/blocktest/main.go:120 +0x359
```

File: cmd/blocktest/main.go [hunk 4]
```diff
@@ -103,103 +96,44 @@ type Test struct {
...
 	}
-	blocks, err := loadBlocksFromTestFile(TestFile)
+	blocks, err := loadBlocksFromTestFile(os.Args[1])
 	if err != nil {
-		panic(err)
+		utils.Fatalf("Could not load blocks: %v", err)
 	}
 
-	runtime.GOMAXPROCS(runtime.NumCPU())
+	chain := memchain()
```
Comment (Gustav-Simonsson) [inline cmd/blocktest/main.go]: We also need RPC API started, as we want the test to end with enough running to allow ethereum.js to call RPC API to test stuff.

File: xeth/xeth.go (hunk 3 above)
Comment (fjl) [inline xeth/xeth.go]: This branch is taken if `UnlockAccount` returned true but `Sign` still returns `ErrLocked` after that.

**Pull request merged**